x-pack/plugins/security/server/routes/profile/update.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import type { RouteDefinitionParams } from '../';
+import { wrapIntoCustomErrorResponse } from '../../errors';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+
+export function defineUpdateProfileDataRoute({ router, getProfileService }: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/security/profile/_data/{uid}',
+      validate: {
+        params: schema.object({ uid: schema.string() }),
+        body: schema.recordOf(schema.string(), schema.any()),
```

Comment:
__question:__ `schema.any()` - do we also support nested objects or we can limit that to `number | string | boolean`?

Reply:
Yeh, this data could literally be anything including nested so don't think we can restrict further

Reply:
Got it, thanks for confirming